en/developer/appdev/articles/03-service-builder/04-business-logic-with-service-builder/02-creating-local-services/03-implementing-methods-to-get-and-count-entities.markdown

```diff
@@ -127,12 +127,11 @@ public int getGuestbookEntriesCount(long groupId, long guestbookId) {
 Now your local service can get entities matching your criteria and return quick
 entity counts. 
 
-## Service Method Prefixes and Database Transactions
+## Service Method Prefixes and AOP Transactions
 
-@product@ uses read-only database transactions in some methods. Specific service
-method name prefixes determine whether read-only database transactions are used.
-Service methods prefixed with any of these words precipitate read-only
-transactions: 
+Liferay's AOP framework applies database transactions to service methods based
+on the method name prefix. Service methods prefixed with any of these words
```

Comment:
We need to highlight this prefix thing only controls default behavior, which can be explicit override, then whatever the name says won't matter anymore.

Reply:
@shuyangzhou , what is it that someone would override that would change whether a read-only transaction is applied?

Reply:
Every tx attribute can be overidded, see https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/transaction/Transactional.java

Reply:
yes, the annotation scanning is from bottom to top, the first seen wins. So it means the one in subclass will override the one in super. And method level one will override the class level one.